### PR TITLE
Improved Code

### PR DESCRIPTION
**Pull request by prvtfly**
- Promised Based Database Querys
- Fixed Linter Errors
- Fixed Nestiness in this code
- Additonal Error Handling and a Function in the Config which can handle Cheater punishment
- Better naming

### CODE REVIEW

**Comment by prvtfly** [comment]
awaiting pr

File: client/main.lua [hunk 1]
```diff
@@ -150,9 +150,8 @@ function OpenLicenceMenu(shop)
 		ESX.OpenContext("right", elements2, function(menu2,element2)
 			-- If the value is no, close the menu and stop the script. 
 			if element2.val == "no" then
-                ESX.CloseContext()
-                return;
-            end
+        return ESX.CloseContext();
```
Comment (Gellipapa) [inline client/main.lua:153]: remove please ';' all code

File: client/main.lua [hunk 2]
```diff
@@ -197,9 +196,11 @@ CreateThread(function()
 	while true do
 		Wait(0)
 
-		if isInShopMenu then
-			DisableControlAction(0, 75, true)  -- Disable exit vehicle
-			DisableControlAction(27, 75, true) -- Disable exit vehicle
+    local VEH_EXIT_CONTROL = 75;
```
Comment (Gellipapa) [inline client/main.lua:199]: Indent fix please

File: client/marker.lua [hunk 3]
```diff
@@ -51,23 +55,27 @@ AddEventHandler('esx_boat:hasEnteredMarker', function(zone, zoneNum)
...
-				CurrentAction     = 'garage_in'
-				CurrentActionMsg  = TranslateCap('garage_store')
-				CurrentActionData = { vehicle = vehicle, zoneNum = zoneNum }
-			end
+
+		if not IsPedInAnyVehicle(playerPed, false) then
+      return;
 		end
+
+		local vehicle = GetVehiclePedIsIn(playerPed, false)
```
Comment (Gellipapa) [inline client/marker.lua:63]: indent please

File: server/main.lua [hunk 4]
```diff
@@ -1,87 +1,101 @@
...
+    return true;
+  end
+
+  print(('[^2INFO^7] Stored ^5%s^7 %s !'):format(affectedRows, affectedRows > 1 and 'boats' or 'boat'))
+  return true;
 end
 
 MySQL.ready(function()
-	ParkBoats()
+	local storedBoats = ParkBoats()
```
Comment (Gellipapa) [inline server/main.lua:15]: indent please

File: server/main.lua [hunk 5]
```diff
@@ -1,87 +1,101 @@
...
 		['@owner']  = xPlayer.identifier,
 		['@plate']  = plate
-	}, function(rowsChanged)
-		if rowsChanged == 0 then
-			print(('[^2INFO^7] Player ^5%s^7 Attempted To Exploit Garage'):format(xPlayer.source))
-		end
-	end)
+	});
+
+  if affectedRows > 0 then
```
Comment (Gellipapa) [inline server/main.lua:64]: indent please, format code.

**Review by Gellipapa — CHANGES_REQUESTED** [review]
Hi! Thank you pr, please handle requests.

**Review by Gellipapa — APPROVED** [review]
Hi! Well done, I'll test and merge.

**Comment by Gellipapa** [comment]
@prvtfly Hi! This is not working i can't store boat, all time gettting "This is not my boat", i tested old version works correctly. I think you mistake in server side querys. Please fix this issue and test. Thanks.

**Comment by Gellipapa** [comment]
@prvtfly Hi! Can you fix issues?

**Comment by prvtfly** [comment]
Hey, I’ll have a look tomorrow!

**Comment by Gellipapa** [comment]
> Hey, I’ll have a look tomorrow!

Thanks 😄

**Pull request closed without merging**